apps/io-wallet-user-func/src/infra/azure/cosmos/wallet-instance.ts

```diff
@@ -177,6 +177,63 @@ export class CosmosDbWalletInstanceRepository
     );
   }
 
+  getNotRevokedByDiffirentIdAndUserId(
+    id: WalletInstance["id"],
+    userId: WalletInstance["userId"],
+  ) {
+    return pipe(
+      TE.tryCatch(
+        async () => {
+          const { resources: items } = await this.#container.items
+            .query({
+              parameters: [
+                {
+                  name: "@partitionKey",
+                  value: userId,
+                },
+                {
+                  name: "@idKey",
+                  value: id,
+                },
+              ],
+              query:
+                "SELECT * FROM c WHERE c.id != @idKey AND c.userId = @partitionKey AND c.isRevoked = false",
+            })
+            .fetchAll();
+          return items;
+        },
+        (error) =>
+          error instanceof Error && error.name === "TimeoutError"
+            ? new ServiceUnavailableError(
+                `The request to the database has timed out: ${error.message}`,
+              )
+            : new Error(`Error getting wallet instances by user id: ${error}`),
+      ),
+      TE.chain((items) =>
+        pipe(
+          items,
+          RA.head,
+          O.fold(
+            () => TE.right(O.none),
+            () =>
+              pipe(
+                items,
+                t.array(WalletInstance).decode,
```

Comment:
since they are all valid, you can use `WalletInstanceValid`

Reply:
If I write `WalletInstanceValid` (instead of `WalletInstance`), spawn a linting/typecheck error like this:

<img width="838" alt="Screenshot 2024-11-08 at 12 30 14" src="https://github.com/user-attachments/assets/e047c8ae-dbde-4d66-aabb-88b3006974dc">

I've imported the `WalletInstanceValid` type from `io-wallet-common/wallet-instance`: is it possible to modify the `io-wallet-common` package? Or, simply, I can disable typecheck with `//@ts-ignore`?

Reply:
yes sure you can